core/src/test/java/androidx/documentfile/provider/AssetsDocumentFile.java

```diff
@@ -0,0 +1,142 @@
+package androidx.documentfile.provider;
+
+import android.content.res.AssetManager;
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.io.IOException;
+
+/**
+ * <p>Wraps an Android assets file or folder as a DocumentFile object.</p>
+ *
+ * <p>This is used to emulate access to the external storage.</p>
+ */
+public class AssetsDocumentFile extends DocumentFile {
+
+    /**
+     * Absolute file path in the assets folder.
+     */
+    @NonNull
+    private final String fileName;
+
+    @NonNull
+    private final AssetManager assetManager;
+
+    public AssetsDocumentFile(@NonNull String fileName, @NonNull AssetManager assetManager) {
+        super(null);
+        this.fileName = fileName;
+        this.assetManager = assetManager;
+    }
+
+    @Nullable
+    @Override
+    public DocumentFile createFile(@NonNull String mimeType, @NonNull String displayName) {
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public DocumentFile createDirectory(@NonNull String displayName) {
+        return null;
+    }
+
+    @NonNull
+    @Override
+    public Uri getUri() {
+        return Uri.parse(fileName);
+    }
+
+    @Nullable
+    @Override
+    public String getName() {
+        int pos = fileName.indexOf('/');
+        if (pos >= 0) {
+            return fileName.substring(pos + 1);
+        } else {
+            return fileName;
+        }
+    }
+
+    @Nullable
+    @Override
+    public String getType() {
+        if (fileName.endsWith(".mp3")) {
+            return "audio/mp3";
+        } else if (fileName.endsWith(".png")) {
+            return "image/png";
```

Comment:
How about using `MimeTypeMap.getMimeTypeFromExtension`? That would work for more file types, just in case we used the class for more things later.

Reply:
Good point. I made the code more generic by using the suggested method.

Note: When using Robolectric the class `MimeTypeMap` is replaced by the class `ShadowMimeTypeMap` that is empty by default. So I had to call the method `addExtensionMimeTypMapping()` to register the `mp3` file extension. I already had that line of code, removed it, and now it is back again. :grin:

Reply:
Oh okay. As you can probably already tell, I have never worked with roboelectric before :) Your change looks good.